code/src/components/PostMessage.js

```diff
@@ -0,0 +1,57 @@
+import React from 'react'
+
+import { HAPPY_THOUGHTS_API } from '../reusable/urls'
+
+const PostMessage = ({ messageNew, setMessageNew, messageList, setMessageList, chars, setChars }) => {
+
+  const onMessageNewChange = (event) => {
+    setMessageNew(event.target.value)
+    setChars(event.target.value.length)
+  }
+
+  const fetchMessageList = () => {
+    fetch(HAPPY_THOUGHTS_API)
+      .then(response => response.json())
+      .then(messages => setMessageList(messages))
+      .catch(err => console.error(err))
+  }
+
+  const onFormSubmit = (event) => {
+    event.preventDefault()
+
+    const postArgument = {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify({ message: messageNew })
+    }
+
+    fetch(HAPPY_THOUGHTS_API, postArgument)
+      .then(response => response.json())
+      .then(() => {
+        setMessageNew('')
+        fetchMessageList()})
+  }
+
+  return (
+    <article className="message-container form-container">
+      <form className="post-message-form" onSubmit={onFormSubmit}>
+        <label htmlFor="newMessage"><h3 className="post-message-label">What's making you happy right now?</h3></label>
```

Comment:
I would consider the usage of h3 here inside of a label tag. I don’t think this is a semantically correct way of using a h3. I would consider using a span tag if you want to wrap the text here in an element 🙂